Improve landing page (#1853)
SAP/SapMachine: index.md

title: SapMachine
---

**SapMachine** is the free, multiplatform, production-ready, and [Java SE certified](https://github.com/SAP/SapMachine/wiki/Certification-and-Java-Compatibility) [OpenJDK](https://openjdk.org/) (Open Java Development Kit) distribution by [SAP](https://sapmachine.io). It serves as the default JDK for SAP's countless applications and services.

> 

Reviewer: 27.4

Author: > Erstmal vorab: vielen vielen Dank für den Fame.
> Die Struktur ist gleich geblieben, die Sätze sind an sich gleich und an ähnlichen Stellen geblieben, ... das bedeutet, dass das ja gut so gewesen sein muss.

Dein ursprünglicher Text war gut. Dein vorheriges Rework der Website sinnvoll, die ursprüngliche Version war voll mit unnötigen Informationen (JCP Mitgliedschaften und mehr). Ich habe ihn mit ein paar Links angereichert (hilft bei SEO), mit dem Text von Coretto abgeglichen und etwas konkretisiert.

Bzgl. JMC: Ich habe ihn jetzt weiter unter erwähnt. Danke für deinen Hinweis.

m) warum geht der Link auf diesem PullRequest mit dem Text "SAP" nicht zu SAP?

Danke für den Hinweis, ist behoben.

<img align="left" width="240" src="assets/images/logo_circular.svg" alt="Logo of SapMachine">

> This Distribution supports all major Operation Systems and CPU architectures, including PowerPC and AIX.

Reviewer 2: Operating Systems.

Author: That's a good point.

SapMachine comes with long-term support releases that include bug fixes and performance updates; you can learn more about support and maintenance in our [wiki](https://github.com/SAP/SapMachine/wiki/Maintenance-and-Support).

Our goal is to keep SapMachine as close to OpenJDK as possible,
only adding additional features when absolutely necessary; you can find a list of these
in the [wiki](https://github.com/SAP/SapMachine/wiki/Differences-between-SapMachine-and-OpenJDK). One major difference is that we started including [async-profiler](https://github.com/jvm-profiling-tools/async-profiler) in our distribution.

> Our team's many contributions include the [PowerPC/AIX support](http://openjdk.java.net/projects/ppc-aix-port/), [helpful NullPointerExceptions](https://openjdk.org/jeps/358), a [website for JFR events](https://sap.github.io/SapMachine/jfrevents/), our own [IntelliJ Profiler plugin](https://plugins.jetbrains.com/plugin/20937-java-jfr-profiler), and more.

Reviewer 2: Suggestion: 'Our team's many contributions to the OpenJDK and its ecosystem'

title: SapMachine
---

> **SapMachine** is the free, multiplatform, production-ready, and [Java SE certified](https://github.com/SAP/SapMachine/wiki/Certification-and-Java-Compatibility) [OpenJDK](https://openjdk.org/) (Open Java Development Kit) distribution by [SAP](https://sap.com). It serves as the default JDK for SAP's countless applications and services.

Reviewer 3: SapMachine is one of many distributions of OpenJDK, so I would prefer something like:

```suggestion
**SapMachine** is a distribution of [OpenJDK](https://openjdk.org/) developed by SAP. It is designed to be free, cross-platform, production-ready, and [Java SE certified](https://github.com/SAP/SapMachine/wiki/Certification-and-Java-Compatibility). This distribution serves as the default Java Runtime for SAP's numerous applications and services.
```

or 

```suggestion
**SapMachine** is a free, cross-platform, production-ready distribution of [OpenJDK](https://openjdk.org/) (Open Java Development Kit) that is [Java SE certified](https://github.com/SAP/SapMachine/wiki/Certification-and-Java-Compatibility) by [SAP](https://sap.com). It serves as the default JDK for SAP's numerous applications and services.
```

Reviewer 3: Oh, and while we are at it..., if you can agree would you mind changing the title, too? [https://github.com/SAP/SapMachine/blob/gh-pages/_config.yml](url)

It currently is:
`The OpenJDK release maintained and supported by SAP`

How about something a little more catchy?
`Fast. Reliable. Open.`

> **SapMachine** is a distribution of [OpenJDK](https://openjdk.org/) developed by <a href="https://sap.com">SAP</a>. It is designed to be free, cross-platform, production-ready, and [Java SE certified](https://github.com/SAP/SapMachine/wiki/Certification-and-Java-Compatibility). This distribution serves as the default Java Runtime for SAP's numerous applications and services.

Reviewer 2: ```suggestion
**SapMachine** is a distribution of the [OpenJDK](https://openjdk.org/) maintained by <a href="https://sap.com">SAP</a>. It is designed to be free, cross-platform, production-ready, and [Java SE certified](https://github.com/SAP/SapMachine/wiki/Certification-and-Java-Compatibility). This distribution serves as the default Java Runtime for SAP's numerous applications and services.
```

Added a "the" and shouldn't we rather write "maintained" instead of "developed"? Because it is fair to say that most of the development of the OpenJDK is done outside our team...

Built and supported by SAP, it provides a robust and reliable Java environment.
**SapMachine** is the free, multiplatform, production-ready, and [Java SE certified](https://github.com/SAP/SapMachine/wiki/Certification-and-Java-Compatibility) [OpenJDK](https://openjdk.org/) (Open Java Development Kit) distribution by [SAP](https://sap.com). It serves as the default JDK for SAP's countless applications and services.

<img align="left" width="240" src="assets/images/logo_circular.svg" alt="Logo of SapMachine">

Following the release cadence of the OpenJDK, a new feature release is shipped every six months, with one designated as a long-term support release every two years.

> Update releases for active versions are provided every quarter. So SapMachine offers long-term support, regular performance improvements and timely security updates.

Reviewer 4: Wir hatten vor nicht all zu langer Zeit (2 Jahre?) diskutiert, ob der Inhalt der maintenance wiki Seite nicht hier repliziert werden sollte, da scheinbar immer wieder Anfragen diesbezüglich kommen, weil Leute nicht dem Link folgen.
Wir hatten beschlossen dass dieser Satz hier reicht da man solchen Leuten eh nicht helfen kann. Wird jetzt halt noch knapper.  Nur um Kontext zu diesem Satz zu geben, mir persoenlich ist das egal.

Aber auf der wiki Seite dort steht falsche Info, da steht dass wir LTEs 4 Jahre supporten, das sollte aber 5 heissen, oder?

Author: > Aber auf der wiki Seite dort steht falsche Info, da steht dass wir LTEs 4 Jahre supporten, das sollte aber 5 heissen, oder?

Könnte. Bitte ändere dass passend.

> This distribution supports all major operating systems.

Reviewer 2: ```suggestion
SapMachine supports all major operating systems.
```

> SapMachine comes with long-term support releases that include bug fixes and performance updates; you can learn more about support and maintenance in our [wiki](https://github.com/SAP/SapMachine/wiki/Maintenance-and-Support).

Reviewer 2: ```suggestion
It comes with long-term support releases that include bug fixes and performance updates; you can learn more about support and maintenance in our [Wiki](https://github.com/SAP/SapMachine/wiki/Maintenance-and-Support).
```
Changed the beginnings of the sentences and capitalized "Wiki"

> only adding additional features when absolutely necessary; you can find a list of these in the [wiki](https://github.com/SAP/SapMachine/wiki/Differences-between-SapMachine-and-OpenJDK).

Reviewer 2: ```suggestion
only adding additional features when absolutely necessary; you can find a list of these in the [Wiki](https://github.com/SAP/SapMachine/wiki/Differences-between-SapMachine-and-OpenJDK).
```

Our team's many contributions to the OpenJDK and the ecosystem include the [PowerPC/AIX support](http://openjdk.java.net/projects/ppc-aix-port/),

>  [helpful NullPointerExceptions](https://openjdk.org/jeps/358), a [website for JFR events](https://sap.github.io/SapMachine/jfrevents/) and [more](https://github.com/SAP/SapMachine/wiki/).

Reviewer 4: Man koennte JEP 387 Elastic Metaspace nennen, und bald auch deinen, Johannes.

Author: Das klingt nach einer guten Idee, ich füge den JEP 387 hinzu.

Leider haben wir keine Liste unserer großen Contributions im Wiki, vielleicht sollte man mal eine List mit diesen anlegen.

>  [helpful NullPointerExceptions](https://openjdk.org/jeps/358), a [website for JFR events](https://sap.github.io/SapMachine/jfrevents/) and [more](https://github.com/SAP/SapMachine/wiki/).

Reviewer 2: The last link (more) should be https://github.com/SAP/SapMachine/wiki/Features-Contributed-by-SAP

Author: No. The page is outdated. We should only link to it when it is maintained again.

title: SapMachine
---

**SapMachine** is the free, multiplatform, production-ready, and [Java SE certified](https://github.com/SAP/SapMachine/wiki/Certification-and-Java-Compatibility) [OpenJDK](https://openjdk.org/) (Open Java Development Kit) distribution by [SAP](https://sap.com). It serves as the default JDK for SAP's countless applications and services.

<img align="left" width="240" src="assets/images/logo_circular.svg" alt="Logo of SapMachine">

> This Distribution supports all major Operation Systems.

Reviewer 2: It's still "Operation Systems" instead of "Operating Systems". Also, "distribution" should not start with a capital letter here.

Author: Good catch :)

SapMachine comes with long-term support releases that include bug fixes and performance updates; you can learn more about support and maintenance in our [wiki](https://github.com/SAP/SapMachine/wiki/Maintenance-and-Support).

Our goal is to keep SapMachine as close to OpenJDK as possible,
only adding additional features when absolutely necessary; you can find a list of these
in the [wiki](https://github.com/SAP/SapMachine/wiki/Differences-between-SapMachine-and-OpenJDK). One major difference is that we started including [async-profiler](https://github.com/jvm-profiling-tools/async-profiler) in our distribution.

> 

Reviewer 2: We should not mention Async Profiler here explicitly but add it to the differences page in the Wiki.

Author: I'll do

Author: I added a preliminary sentence to the wiki.

Our team's many contributions to the OpenJDK and the ecosystem include the [PowerPC/AIX support](http://openjdk.java.net/projects/ppc-aix-port/),
 [helpful NullPointerExceptions](https://openjdk.org/jeps/358), a [website for JFR events](https://sap.github.io/SapMachine/jfrevents/),
 our own [IntelliJ Profiler plugin](https://plugins.jetbrains.com/plugin/20937-java-jfr-profiler).

> 

Reviewer 2: Do not mention the IntelliJ Profiler plugin here because it's rather Johannes Bechberger, not SapMachine 😉

Author: Ok, but we already showed it to all our stakeholders and started to actively recruit someone to work on this plugin. So it's a project of SapMachine.

Author: Also it was part of the original version of the homepage and is prominently presented on the main page of the [wiki](https://github.com/SAP/SapMachine/wiki). So I think it is a SapMachine project. I totally agree that we should move the Plugin from my personal GitHub account to SAP, but that is out of scope for this PR.

Reviewer 2: Yes, it is a SapMachine team project in a sense that we endorse it and you did it and still do it while working in the team. However, I would not mention it here on the front page yet, as long as the plugin is not really owned by sapmachine or SAP in IntelliJ.

Author: I remove it then

Author: I hope it is ok to still mention https://sap.github.io/SapMachine/jfrevents, because it is used by multiple people in our team and it's published on the SapMachine site.
